plugins/radar-android-polar/src/main/java/org/radarbase/passive/polar/PolarManager.kt

```diff
@@ -125,8 +123,7 @@ class PolarManager(
             override fun deviceDisconnected(polarDeviceInfo: PolarDeviceInfo) {
                 logger.debug("Device disconnected ${polarDeviceInfo.deviceId}")
                 isDeviceConnected = false
-                status = SourceStatusListener.Status.DISCONNECTED // red circle
-
+                disconnect()
```

Comment:
I am not entirely sure, but does it not loop between connecting and disconnecting the manager when this callback is received? If so, we could add logic specific to the needs of this plugin ( such as handling the merging of `::disconnect` with the `UNAVAILABLE` status or similar to this) to prevent the loop and starting from READY again during reconnections.

**Please disregard this review if the behavior is functioning correctly with `::disconnect` as well.**

Reply:
I don't think I completely understand; when the device is disconnected, this callback is called, which sets the plugin state to disconnect. The recording is started again, and it becomes ready, looking for the device to connect again.

Here are logs showing the process

```
2024-09-09 16:01:27.858 19539-19539 pRMT:PolarManager       org.radarcns.detail                  D  Device disconnected CF27042A
2024-09-09 16:01:27.859 19539-19539 pRMT:SourceServiceConn* org.radarcns.detail                  I  AppSource status changed of source Polar Vantage V3 CF27042A
2024-09-09 16:01:27.859 19539-19539 pRMT:SourceServiceConn* org.radarcns.detail                  I  Updated source status to DISCONNECTING
2024-09-09 16:01:27.859 19539-19539 pRMT:RadarService       org.radarcns.detail                  I  Source of ServiceConnection<org.radarbase.passive.polar.PolarService> was updated to DISCONNECTING
2024-09-09 16:01:27.874 19539-19539 pRMT:SourceServiceConn* org.radarcns.detail                  I  AppSource status changed of source null
2024-09-09 16:01:27.874 19539-19539 pRMT:SourceServiceConn* org.radarcns.detail                  I  Updated source status to DISCONNECTED
2024-09-09 16:01:27.874 19539-19539 pRMT:RadarService       org.radarcns.detail                  I  Source of ServiceConnection<org.radarbase.passive.polar.PolarService> was updated to DISCONNECTED
2024-09-09 16:01:27.875 19539-19665 pRMT:RadarService       org.radarcns.detail                  I  Starting recording on connection ServiceConnection<org.radarbase.passive.polar.PolarService>
2024-09-09 16:01:27.880 19539-11102 pRMT:SourceService      org.radarcns.detail                  I  Starting recording now for PolarService
2024-09-09 16:01:27.883 19539-11102 pRMT:PolarManager       org.radarcns.detail                  D  Connecting to Polar API
2024-09-09 16:01:27.883 19539-19539 pRMT:SourceServiceConn* org.radarcns.detail                  I  AppSource status changed of source Polar
2024-09-09 16:01:27.883 19539-19539 pRMT:SourceServiceConn* org.radarcns.detail                  I  Updated source status to READY
2024-09-09 16:01:27.884 19539-19539 pRMT:RadarService       org.radarcns.detail                  I  Source of ServiceConnection<org.radarbase.passive.polar.PolarService> was updated to READY

```

I did not notice any looping in my testing.

Reply:
Okay, that’s great. I have observed this looping behavior in other plugins, but I have not worked with any wearable devices for pRMT, so I was not sure about the behavior for wearables on `::disconnect`

Thanks for the logs.